packages/protocol/contracts/governance/VestingInstance.sol

```diff
@@ -0,0 +1,356 @@
+     * @param timestamp the timestamp at which the calculate the releasable amount
+     */
+    function _getReleasableAmount(uint256 timestamp) public view returns (uint256) {
+        return _calculateFreeAmount(timestamp).sub(mCurrentlyReleased);
+    }
+
+    /**
+     * @dev Calculates the amount that has already vested.
+     * @param timestamp the timestamp at which the calculate the already vested amount
+     */
+    function _calculateFreeAmount(uint256 timestamp) private view returns (uint256) {
+        uint256 currentBalance = getGoldToken().balanceOf(address(this));
+        uint256 totalBalance = currentBalance.add(mCurrentlyReleased);
+
+        if (timestamp < mCliffStartTime) {
+            return 0;
+        }
+        if (timestamp >= mVestingStartTime.add( mVestingPeriods.mul(mVestingPeriodSec) ) || mRevoked) {
+            return totalBalance;
+        }
+        
+        uint256 gradient = (timestamp.sub(mVestingStartTime)).div(mVestingPeriodSec);
+        return  ( (currentBalance.mul(gradient)).mul(mVestAmountPerPeriod) ).div(totalBalance);
+    }
+
+    /**
+     * @notice A wrapper func for the lock gold method
+     * @param value the value to gold to be locked
+     * @return True if the transaction succeeds.
+     * @dev To be called only by the beneficiary of the vesting
+     */
+    function lockGold(uint256 value) external onlyBeneficiary returns (bool) {
+
+      // the beneficiary may not lock more than the vesting has currently released
+      uint256 unreleasedAmount = _getReleasableAmount(block.timestamp);
+      require(unreleasedAmount >= value, "Gold Amount to lock must not be less that the currently releasable amount");
+
+      bool success;
+      (success,) = address(getLockedGold()).lock.gas(gasleft()).value(msg.value)();
+      require(success);
+      emit VestingGoldLocked(msg.value, block.timestamp);
+      return success;
+    }
+
+    /**
+     * @notice A wrapper func for the unlock gold method function
+     * @param value the value to gold to be unlocked for the vesting instance
+     * @return True if the transaction succeeds.
+     * @dev To be called only by the beneficiary of the vesting
+     */
+    function unlockGold(uint256 value) external onlyBeneficiary returns (bool) {
+      bool success;
+      (success,) = address(getLockedGold()).unlock.gas(gasleft()).value(msg.value)(value);
+      require(success);
+      emit VestingGoldUnlocked(msg.value, block.timestamp);
+      return success;
+    }
+
+    /**
+     * @notice A wrapper func for the relock locked gold method function
+     * @param index the index of the pending locked gold withdrawal
+     * @return True if the transaction succeeds.
+     * @dev To be called only by the beneficiary of the vesting.
+     */
+    function relockLockedGold(uint256 index) external onlyBeneficiary returns (bool) {
+      bool success;
+      (success,) = address(getLockedGold()).relock.gas(gasleft())(index);
```

Comment:
This line should be `getLockedGold().relock(index);`

Ditto for all the other wrapped calls, unless they return booleans, in which case we should require the return value is true

Reply:
done